woocommerce/woocommerce-android
# Order creation - products multi-selection

### Final PR for _Speedy IPP: Product Multi-Select_ M1.

### Description

This PR integrates the product selector component into the order creation (and edition) flow.

The goal of this feature is to allow a multi-selection of products while creating an order. 

Previously, the user could select only a single product at once and needed to wait a couple of seconds for the order to get updated each time. Now, the user can select multiple products at once and the order creation flow is significantly faster.

More details in the project thread: pdfdoF-2lX-p2

This PR is based on the following ones, already reviewed:
* https://github.com/woocommerce/woocommerce-android/pull/8545
* https://github.com/woocommerce/woocommerce-android/pull/8553
* https://github.com/woocommerce/woocommerce-android/pull/8519
* https://github.com/woocommerce/woocommerce-android/pull/8443

### Implementation
* The `ProductSelectorFragment` was plugged into the order creation/edition screen and is opened on "+ Add products" button tap.
* Old product picker code was removed
* ProductSelector was improved to allow filtering out draft products and variable products with no variations. This was designed to be configurable, e.g. ProductSelector can still be used in other places and allow selecting draft products. The coupon edition flow was updated as well and doesn't show draft products and variable products without variations.

### Testing instructions

Ensure it's possible to select multiple products and product variations at once while creating new order or editing an existing one (the existing order must not be in a completed state).

### Images/gif



https://user-images.githubusercontent.com/4527432/225295430-1f71e505-77be-48f3-b42a-7b48d997dc7d.mp4
